client/v2/autocli/query.go

```diff
@@ -131,14 +131,14 @@ func (b *Builder) BuildQueryMethodCommand(descriptor protoreflect.MethodDescript
 			return err
 		}
 
-		_, err = fmt.Fprintln(cmd.OutOrStdout(), string(bz))
+		err = b.outOrStdoutFormat(cmd, bz)
 		return err
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	if b.AddQueryConnFlags != nil {
+	if cmd != nil && b.AddQueryConnFlags != nil {
```

Comment:
just curious why cmd == nil and err == nil is a valid case, which requires you to have that extra check

Reply:
If options has skip it returns nil in cmd which may lead to a nil pointer  if we try to add flags in it
`if options.Skip {
		return nil, nil
	}`

Reply:
However, I see even `BuildMsgMethodCommand` and `BuildQueryMethodCommand` returns nil command and nil error. I just find that weird, but nvm then 🤷🏾‍♂️

Reply:
Yeah, I like how it sounds. Let me explore the idea. Also idk I'm always afraid when a value that can be nil is not checked

Reply:
```go
	for i := 0; i < n; i++ {
		methodDescriptor := methods.Get(i)
		methodOpts, ok := rpcOptMap[methodDescriptor.Name()]
		if !ok {
			methodOpts = &autocliv1.RpcCommandOptions{}
		}

		if methodOpts.Skip {
			continue
		}

		methodCmd, err := b.BuildQueryMethodCommand(methodDescriptor, methodOpts)
		if err != nil {
			return err
		}

		cmd.AddCommand(methodCmd)
	}
```
Personally something like this makes more sense to me, than

```go
		methodDescriptor := methods.Get(i)
		methodOpts := rpcOptMap[methodDescriptor.Name()]

		methodCmd, err := b.BuildQueryMethodCommand(methodDescriptor, methodOpts)
		if err != nil {
			return err
		}

		if methodCmd != nil {
			cmd.AddCommand(methodCmd)
		}
```

and the following in `buildMethodCommandCommon`.

```go
	if options == nil {
		// use the defaults
		options = &autocliv1.RpcCommandOptions{}
	}

	if options.Skip {
		return nil, nil
	}
```

Reply:
I like this!!